the-collab-lab/tcl-50-smart-shopping-list
# Sort items by how soon they are needed

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
In this new feature, users are able to sort items by how soon they might need to buy them.  A user can easily identify what they need to buy, which will in turn help them better plan their shopping trips. 



## Related Issue
Closes #12 


## Acceptance Criteria


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before
![image](https://user-images.githubusercontent.com/57120467/201055245-2163bf52-7c2b-4f80-a08e-45b24873ad17.png)



### After
![image](https://user-images.githubusercontent.com/57120467/201226982-0e93fd6e-fdf0-4572-b50f-95eda051d0be.png)




## Testing Steps / QA Criteria
1. From your terminal, do `git checkout main`
2. Pull down this branch with `git pull origin kl-md-sorting-items`
3. Checkout the branch with `git checkout kl-md-sorting-items`
4. Run `npm start` to view on the browser
5. To test this feature: you need to join an existing list or create a new list and add items, then click on the list link to view your listed items.
6. Click to check any of the items you want to purchase
